File: order-producer/src/main/java/com/sahakav/coffee_shop/event/handler/CoffeeEventHandler.java
package com.sahakav.coffee_shop.event.handler;

import com.sahakav.coffee_shop.event.CreateCoffeeOrderEvent;
import com.sahakav.coffee_shop.service.EventsProducer;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

@Service
public class CoffeeEventHandler {

    @Autowired
    private EventsProducer eventsProducer;

    public void handle(CreateCoffeeOrderEvent createCoffeeOrderEvent){
       eventsProducer.produce(createCoffeeOrderEvent);}
}

File: order-producer/src/main/scala/com/sahakav/coffee_shop/event/handler/CoffeeEventHandler.scala
package com.sahakav.coffee_shop.event.handler

import java.util.UUID

import com.sahakav.coffee_shop.event.CreateCoffeeOrderEvent
import com.sahakav.coffee_shop.service.EventsProducer
import org.springframework.stereotype.Service

@Service
class CoffeeEventHandler(val eventsProducer: EventsProducer) {

  def handle(createCoffeeOrderEvent: CreateCoffeeOrderEvent): UUID = {
    eventsProducer.produce(createCoffeeOrderEvent)
    createCoffeeOrderEvent.orderId
  }
}
